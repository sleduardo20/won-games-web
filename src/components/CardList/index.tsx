import GameItem, { GameItemProps } from '../GameItem';
import { Container, Footer, Total } from './styles';

export interface CardListProps {
  items: GameItemProps[];
  total: string;
}

const CardList = ({ items, total }: CardListProps) => {
  return (
    <Container>
      {items.map(item => (
        <GameItem key={item.title} {...item} />
      ))}

      <Footer>
        Total <Total>{total}</Total>
      </Footer>
    </Container>
  );
};

export default CardList;
