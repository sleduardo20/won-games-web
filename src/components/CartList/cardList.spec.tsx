import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/tests/helpers';

import CardList from '.';
import mockCardList from './mock';

describe('<CardList />', () => {
  it('should be able render correctly', () => {
    const { container } = renderWithTheme(
      <CardList items={mockCardList} total="R$ 330,00" />,
    );

    expect(screen.getAllByRole('heading', { level: 3 })).toHaveLength(2);
    expect(screen.getByText('R$ 330,00')).toHaveStyle({ color: '#f231a5' });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should be able render the button', () => {
    renderWithTheme(
      <CardList hasButton items={mockCardList} total="R$ 330,00" />,
    );

    expect(screen.getByText(/Buy it now/)).toBeInTheDocument();
  });

  it('should be able render empty if there are no games', () => {
    renderWithTheme(<CardList />);

    expect(screen.getByText(/your cart is empty/i)).toBeInTheDocument();
    expect(screen.queryByText('total')).not.toBeInTheDocument();
  });
});
