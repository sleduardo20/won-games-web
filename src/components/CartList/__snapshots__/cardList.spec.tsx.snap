// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardList /> should be able render correctly 1`] = `
.c2 {
  padding: 2.4rem;
  border-bottom: 0.1rem solid #EAEAEA;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
}

.c4 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 1.2rem;
  width: 9.6rem;
  height: 5.6rem;
  position: relative;
}

.c4 img {
  object-fit: cover;
}

.c5 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c6 {
  font-size: 1.4rem;
  line-height: 1.4rem;
  font-weight: 600;
  color: #030517;
  margin-bottom: 0.8rem;
}

.c8 {
  color: #fafafa;
  padding: 0.2rem 0.8rem;
  background-color: #3cd3c1;
  border-radius: 0.4rem;
  font-size: 1.4rem;
  font-weight: 600;
}

.c0 {
  background: #fafafa;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-self: start;
  -ms-flex-item-align: start;
  align-self: start;
}

.c9 {
  background: #eee;
  color: #030517;
  font-weight: 600;
  font-size: 1.4rem;
  padding: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c10 {
  color: #f231a5;
}

.c1 {
  max-height: 40rem;
  overflow-y: auto;
}

@media (min-width:768px) {
  .c2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:768px) {
  .c4 {
    margin-right: 1.6rem;
    width: 15rem;
    height: 7rem;
  }
}

@media (min-width:768px) {
  .c6 {
    font-size: 2rem;
    line-height: 2rem;
  }
}

@media (min-width:768px) {
  .c9 {
    font-size: 1.6rem;
    padding: 2.4rem;
  }
}

<div
  class="c0"
  data-cy="cart-list"
>
  <div
    class="c1"
  >
    <div
      class="c2"
      data-cy="game-item"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <img
            alt="Red Dead Redemption 2"
            layout="fill"
            src="https://cdn2.unrealengine.com/Diesel%2Fproductv2%2Fheather%2Fhome%2FEGS_RockstarGames_RedDeadRedemption2_G1A_00-1920x1080-308f101576da37225c889173094f373f2afc56c1.jpg?h=1080&resize=1&w=1920"
          />
        </div>
        <div
          class="c5"
        >
          <h3
            class="c6"
          >
            Red Dead Redemption 2
          </h3>
          <div
            class="c7"
          >
            <p
              class="c8"
            >
              R$ 215,00
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c2"
      data-cy="game-item"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <img
            alt="Borderlands 3"
            layout="fill"
            src="https://cdn2.unrealengine.com/Diesel%2Fproduct%2Fcatnip%2FEGS_2K_OAK_IC1_Logo_Transparent_400x800-800x400-847db268641618b804c2dc3e64f0055387fd10e5.png?h=270&amp;resize=1&amp;w=480"
          />
        </div>
        <div
          class="c5"
        >
          <h3
            class="c6"
          >
            Borderlands 3
          </h3>
          <div
            class="c7"
          >
            <p
              class="c8"
            >
              R$ 215,00
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="c9"
  >
    <span>
      Total:
    </span>
    <span
      class="c10"
    >
      R$ 330,00
    </span>
  </div>
</div>
`;
